pkg/apis/pipeline/v1alpha1/taskrun_types.go

```diff
@@ -68,6 +68,10 @@ type TaskRunSpec struct {
 	// If specified, the pod's scheduling constraints
 	// +optional
 	Affinity *corev1.Affinity `json:"affinity,omitempty"`
+	// Used to add environment variables configured on the Task or its steps. Will be overridden by Task and step
+	// environment variables if they have the same name.
+	// +optional
+	AdditionalEnv []corev1.EnvVar `json:"additionalEnv,omitempty"`
```

Comment:
I was avoiding `Env` because it felt a bit weird to have `Env` on `Task` and `Env` on `TaskRun` - I wanted to create a distinction between them. But it doesn't matter much to me. =)